src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -117,6 +190,54 @@ public Uri GetEndpointForClient(ConfigurationClient client)
             ConfigurationClientWrapper currentClient = _clients.FirstOrDefault(c => c.Client == client);
             
             return currentClient?.Endpoint;
+        }
+
+        private async Task RefreshFallbackClients(CancellationToken cancellationToken)
+        {
+            IEnumerable<SrvRecord> results = await _srvLookupClient.Value.QueryAsync(_endpoint.DnsSafeHost, cancellationToken).ConfigureAwait(false);
+
+            // Shuffle the results to ensure hosts can be picked randomly.
+            // Srv lookup may retrieve trailing dot in the host name, just trim it.
+            IEnumerable<string> srvTargetHosts = results.Shuffle().Select(r => $"{r.Target.Value.Trim('.')}").ToList();
+
+            // clean up clients in case the corresponding replicas are removed.
+            foreach (ConfigurationClientWrapper client in _clients)
+            {
+                if (IsEligibleToRemove(srvTargetHosts, client))
+                {
+                    _clients.Remove(client);

```

Comment:
I'd like to keep the static and dynamic clients separate.

The static clients always get invoked first, and they are unchanging. Tracking the two lists separately should help on that.

Reply:
When managing dynamic clients, it can't avoid checking the static clients. Static clients should always prior to the dynamic clients in the first, no need to separate the invocation. I'm fan of keep them in single collection.